phpstan-baseline.neon

```diff
@@ -2749,3 +2749,9 @@ parameters:
 			message: "#^Expected domain exception class Shopware\\\\Core\\\\Framework\\\\Notification\\\\NotificationException, got Shopware\\\\Core\\\\Framework\\\\Api\\\\ApiException$#"
 			count: 2
 			path: src/Core/Framework/Notification/Api/NotificationController.php
+
+		-
+			message: '#^Parameter \#1 \$event of method Shopware\\Core\\Content\\Category\\Subscriber\\CategorySubscriber\:\:categoryLoaded\(\) expects Shopware\\Core\\Framework\\DataAbstractionLayer\\Event\\EntityLoadedEvent\<Shopware\\Core\\Content\\Category\\CategoryEntity\>, Shopware\\Core\\System\\SalesChannel\\Entity\\SalesChannelEntityLoadedEvent\<Shopware\\Core\\Content\\Category\\SalesChannel\\SalesChannelCategoryEntity\> given\.$#'
```

Comment:
please do not add something to the baseline. try to fix the error instead. if it is really not possible, make an inline ignore and with a proper comment why it is not possible to fix this error

Reply:
I would split that method up. it is also better readable.

Reply:
Mh, yes, I will look into that once more.

But what exactly do you mean by split up the method? I already created two test methods, one for the `category.loaded` event and one for the `sales_channel.category.loaded` event :thinking:

Reply:
I adjusted it, to explicitly tell in the `EntityLoadedEvent` that we need something which is covariant to `CategoryEntity` which should fix the issue: https://phpstan.org/blog/whats-up-with-template-covariant#call-site-variance